containers/src/Kokkos_DynRankView.hpp

```diff
@@ -449,10 +449,15 @@ class DynRankView : private View<DataType*******, Properties...> {
   using reference_type = typename view_type::reference_type;
   using pointer_type   = typename view_type::pointer_type;
 
-  using scalar_array_type           = value_type;
-  using const_scalar_array_type     = const_value_type;
-  using non_const_scalar_array_type = non_const_value_type;
-  using specialize                  = typename view_type::specialize;
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
+  using scalar_array_type KOKKOS_DEPRECATED_WITH_COMMENT(
+      "Use data_type instead .") = value_type;
+  using const_scalar_array_type KOKKOS_DEPRECATED_WITH_COMMENT(
+      "Use const_data_type instead .") = const_value_type;
+  using non_const_scalar_array_type KOKKOS_DEPRECATED_WITH_COMMENT(
+      "Use non_const_data_type instead .") = non_const_value_type;
```

Comment:
It's surprising to use a different alias than the one suggested in the deprecation warning. Can you clarify what's going on?
Was this just totally unconforming to the `View` example in https://github.com/kokkos/kokkos/issues/7350#issue-2532348150.

Reply:
I understand your point. I think the original code was wrong and should have been data_type (vs. value_type). So we suggest the use of the correct type which is the data_type.

Reply:
...but we are still using the wrong type?

Reply:
Yes. This would be a drive-by change.

Reply:
So right now we have
```C++
using non_const_value_type = float;
using non_const_scalar_array_type = float*[5];
using non_const_data_type = float*[5];
```
but we want people to use
```C++
float*[5];
```
although this is currently
```C++
float
```
?

Reply:
Nop, the deprecation comment should recommend value_type, is that what you have in mind?

Reply:
The current definiton should match the recommendation so `value_type` seems more appropriate than `data_type`.